packages/firestore/src/core/sync_engine.ts

```diff
@@ -424,27 +424,29 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
     error?: FirestoreError
   ): Promise<void> {
     this.assertSubscribed('applyBatchState()');
-    const mutationBatchResult = await this.localStore.lookupLocalWrite(batchId);
-    assert(
-      mutationBatchResult !== null,
-      'Unable to find mutation batch: ' + batchId
-    );
+    const documents = await this.localStore.lookupMutationDocuments(batchId);
+
+    if (documents === null) {
+      log.debug(LOG_TAG, 'Cannot apply mutation batch with id: ' + batchId);
+      return;
+    }
 
     if (batchState === 'pending') {
       // If we are the primary client, we need to send this write to the
       // backend. Secondary clients will ignore these writes since their remote
       // connection is disabled.
       await this.remoteStore.fillWritePipeline();
     } else if (batchState === 'acknowledged' || batchState === 'rejected') {
-      // NOTE: Both these methods are no-ops for batches that originated from
-      // other clients.
+      // NOTE: These methods are no-ops for batches that originated from other
+      // clients.
```

Comment:
rework this comment if you end up tracking mutation keys for other clients...

Reply:
I moved this line below and added a space to indicate that the comment doesn't apply.